app/controllers/hosts_controller.rb

```diff
@@ -60,6 +60,9 @@ def index(title = nil)
       end
       format.yaml { render :text => search.all(:select => "hosts.name").map(&:name).to_yaml }
       format.json
+      format.csv do
+        send_data search.to_csv(columns: [:id, :name, :os, :environment, :model, :hostgroup, :last_report])
```

Comment:
The default if no columns are passed is to use `column_names` which just gets all the columns from the database. Here I tried to match the UI table, but we don't have to. However `column_names` doesn't provide stuff like `hostgroup`, only `hostgroup_id` - this way we can call methods besides the data that is stored in the db.
I can add the filename export, sure.